lib/index.js

```diff
@@ -228,7 +228,20 @@ function Postgres(a, b) {
   }
 
   function getConnection(reserve) {
-    const connection = slots ? createConnection(options) : connections.shift()
+    let connection
+    while (1) {
+      connection = slots ? createConnection(options) : connections.shift()
+      if (!connection || !connection.hasExpired()) {
+        break
+      }
+
+      // has expired
+      connection.release()
+      const idx = all.findIndex(x => x === connection)
```

Comment:
there is a bug here. `connection` is in a closure and the variable is outside the loop.